scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,579 @@
+        opt.setArgName(PWD_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(GROUPS_OPTION, GROUPS_OPTION_NAME, true,
+                "A comma-separated list of groups the user must be member of. Can be used when the user is created or updated");
+        opt.setArgName(GROUPS_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        optionGroup.setRequired(false);
+        opt = new Option(KEYFILE_OPTION, KEYFILE_OPTION_NAME, true, "Public key path on the local filesystem [default:" +
+                getPublicKeyFilePath() + "]");
+        opt.setArgName(KEYFILE_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(LOGINFILE_OPTION, LOGINFILE_OPTION_NAME, true,
+                "Path to the login file in use [default:" + loginFilePath + "]");
+        opt.setArgName(LOGINFILE_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(GROUPFILE_OPTION, GROUPFILE_OPTION_NAME, true,
+                "Path to the group file in use [default:" + groupFilePath + "]");
+        opt.setArgName(GROUPFILE_OPTION_NAME.toUpperCase());
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = null;
+        try {
+            cmd = parser.parse(options, args);
+        } catch (Exception e) {
+            exitWithErrorMessage(newline + e.getMessage() + newline, "type -h or --help to display help screen", null);
+        }
+        return cmd;
+    }
+
+    private static void updateUserGroups(String login, String[] groups, Multimap<String, String> groupsMap) {
+        groupsMap.removeAll(login);
+        for (String group : groups) {
+            groupsMap.put(login, group);
+            System.out.println("Added group " + group + " to user " + login);
+        }
+    }
+
+    private static void updateUserPassword(PublicKey pubKey, String login, String password, Properties props) throws KeyException {
+        String encodedPassword;
+        encodedPassword = HybridEncryptionUtil.encryptStringToBase64(password, pubKey, FileLoginModule.ENCRYPTED_DATA_SEP);
+        props.put(login, encodedPassword);
+    }
+
+    private static Multimap<String, String> loadGroups(String groupFilePath) throws ManageUsersException {
+        try {
+            Multimap<String, String> groupsMap = TreeMultimap.create();
+            String line = null;
+            BufferedReader reader = new BufferedReader(new FileReader(groupFilePath));
+            try {
+                while ((line = reader.readLine()) != null) {
+                    if (!line.trim().isEmpty()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)